examples/apple/README.md

```diff
@@ -0,0 +1,55 @@
+
+# Apple Login
+
+Login with Apple allows users to login to any web app with their Apple account.
+
+## Web
+
+Package `gologin` provides Go handlers for the Apple OAuth2 Authorization flow and for obtaining the Apple [User struct](https://github.com/dghubble/gologin/blob/master/apple/verify.go).
+
+### Getting Started
+
+    go get github.com/dghubble/gologin/apple
+    cd $GOPATH/src/github.com/dghubble/gologin/examples/apple
+    go get .
+
+## Example App
+
+[main.go](main.go) shows an example web app which uses `gologin` to issue a client-side cookie session. For simplicity, no data is persisted.
+
+Login to your Apple Developer account and navigate to [Certificates, IDs and Profiles](https://developer.apple.com/account/resources/identifiers/list)
```

Comment:
Yes and yes. :-(

I noticed you recently removed the Facebook example because it requires a Facebook account.  I'm guessing that most providers require the creation of a developer account, and that requirements for creating a developer account will vary by provider.  On one hand I can understand that some providers make it much more difficult for the author to validate that they are working, on the other hand having examples is incredibly valuable to developers who are learning how to use gologin.  This is especially true for e.g., Apple where the provided client configs are a bit different from other providers and so some additional responsibility lies with the developer.

What's the rule on whether examples for a provider should be provided in gologin?

Reply:
Yeah, its somewhat problematic. If I'm going to have an example in my library, I'd like to be able to verify it works without paying Apple. Its not a hard rule I have per-se, maybe this has some disclaimer at the top that its unofficial or something.

Reply:
If you provide the Apple handlers in a package of your own, the example might live there. Since it sounds like you're all setup for that.

Maybe the question now is actually, what is the aim in merging this into gologin vs just having your own applelogin package or something that imports gologin? Could be an option if you wanted.